x-pack/filebeat/input/httpjson/split_test.go

```diff
@@ -624,6 +624,83 @@ func TestSplit(t *testing.T) {
 				{"@timestamp": "1234567890", "other_items": "Line 3"},
 			},
 		},
+		{
+			name: "Array of Strings with keep_parent",
+			config: &splitConfig{
+				Target:     "body.alerts",
+				Type:       "array",
+				KeepParent: true,
+			},
+			ctx: emptyTransformContext(),
+			resp: transformable{
+				"body": mapstr.M{
+					"this": "is kept",
+					"alerts": []interface{}{
+						"test1",
+						"test2",
+						"test3",
+					},
+				},
+			},
+			expectedMessages: []mapstr.M{
+				{
+					"this":   "is kept",
+					"alerts": "test1",
+				},
+				{
+					"this":   "is kept",
+					"alerts": "test2",
+				},
+				{
+					"this":   "is kept",
+					"alerts": "test3",
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "Array of Arrays with keep_parent",
+			config: &splitConfig{
+				Target:     "body.alerts",
+				Type:       "array",
+				KeepParent: true,
+			},
+			ctx: emptyTransformContext(),
+			resp: transformable{
+				"body": mapstr.M{
+					"this": "is kept",
+					"alerts": []interface{}{
+						[]interface{}{"test1-1", "test1-2"},
+						[]string{"test2-1", "test2-2"},
+						[]int{1, 2},
+					},
+				},
+			},
+			expectedMessages: []mapstr.M{
+				{
+					"this": "is kept",
```

Comment:
@efd6 my understanding from this test is we are splitting till upto 1 level .. and not recursively any further down ? Is that out of scope for this fix ?

Reply:
That's right.